### ISSUE TITLE
Cleanup heap decorators

### ISSUE THREAD

**Issue by gsingh93** [pwndbg/pwndbg#1208]
The decorators we're using are a bit redundant, for example:
```python
@pwndbg.commands.OnlyWhenRunning
@pwndbg.commands.OnlyWithResolvedHeapSyms
@pwndbg.commands.OnlyWhenHeapIsInitialized
@pwndbg.commands.OnlyWithTcache
```

- If the heap is initialized, then we can assume we're running the target binary.
- If a command requires the tcache to exist, then it should probably mean the heap is initialized
- I'm not sure if we can combine `OnlyWhenHeapIsInitialized` and `OnlyWithResolvedHeapSyms`. Do we ever use symbols before the heap is initialized? If not, then one can imply the other in this case as well.

**Comment by dmur1** [comment]
pull request doesn't contain the bit you were unsure about because i'm not sure about it either

> I'm not sure if we can combine OnlyWhenHeapIsInitialized and OnlyWithResolvedHeapSyms. Do we ever use symbols before the heap is initialized? If not, then one can imply the other in this case as well.

take a look and see what you think

**Comment by gsingh93** [comment]
> Do we ever use symbols before the heap is initialized?

@disconnect3d @lebr0nli @CptGibbon do you know the answer to this?

**Comment by lebr0nli** [comment]
> do you know the answer to this?

I think for `arena` and `mp` commands (which relied on `main_arena`, `thread_arena`, and `mp_` symbols), they probably should be allowed to be used before the heap is initialized since these symbols should already be there even the heap is not initialized, and we are just printing the content of them when executing the commands.
So maybe we could remove that `OnlyWhenHeapIsInitialized` for them.

**Comment by gsingh93** [comment]
Fixed in #1736 (leaving the heap syms one as is for now is fine).
